schema/schema.py

```diff
@@ -0,0 +1,84 @@
+"""Handles JSON schema generation logic"""
+
+
+import click
+import importlib
+
+from samcli.cli.command import _SAM_CLI_COMMAND_PACKAGES
+from samcli.commands._utils.click_mutex import ClickMutex
+from samcli.commands.build.click_container import ContainerOptions
+from samcli.lib.config.samconfig import SamConfig
+
+
+def retrieve_command_structure(package_name: str) -> dict:
+    """Given a SAM CLI package name, retrieve its structure.
+
+    Returns
+    -------
+    dict
+        A dictionary that maps the name of the command to its relevant click options.
+    """
+    module = importlib.import_module(package_name)
+    command_name = package_name.split(".")[-1]  # command name is the last folder
+    command = {}
+
+    def format_param(param: click.core.Option):
+        """Format a click Option parameter to a dictionary object."""
+        formatted_param = {"name": param.name, "help": param.help}
+
+        if param.type.name in ["text", "path", "choice"]:
+            formatted_param["type"] = "string"
+        else:
+            formatted_param["type"] = param.type.name
+
+        if param.default:
+            formatted_param["default"] = param.default
+
+        if param.type.name == "choice":
+            formatted_param["choices"] = param.type.choices
+
+        return formatted_param
+
+    if isinstance(module.cli, click.core.Group):  # command has subcommands (e.g. local invoke)
```

Comment:
We could, though it would involve writing out a whole separate function for parsing one click command. Still though, it would also address some of the code duplication, so it would definitely be a good addition 👍